src/vulkan-renderer/wrapper/resource_descriptor.cpp

```diff
@@ -0,0 +1,110 @@
+#include "inexor/vulkan-renderer/wrapper/resource_descriptor.hpp"
+
+#include "inexor/vulkan-renderer/wrapper/info.hpp"
+
+#include <spdlog/spdlog.h>
+
+#include <cassert>
+
+namespace inexor::vulkan_renderer::wrapper {
+
+ResourceDescriptor::ResourceDescriptor(ResourceDescriptor &&other) noexcept
+    : m_device(other.m_device), m_name(std::move(other.m_name)), m_descriptor_pool(std::exchange(other.m_descriptor_pool, nullptr)),
+      m_descriptor_set_layout(std::exchange(other.m_descriptor_set_layout, nullptr)),
+      m_descriptor_set_layout_bindings(std::move(other.m_descriptor_set_layout_bindings)),
+      m_write_descriptor_sets(std::move(other.m_write_descriptor_sets)),
+      m_descriptor_sets(std::move(other.m_descriptor_sets)), m_swapchain_image_count(other.m_swapchain_image_count) {}
+
+ResourceDescriptor::ResourceDescriptor(VkDevice device, const std::uint32_t swapchain_image_count,
+                                       const std::initializer_list<VkDescriptorType> pool_types,
+                                       std::vector<VkDescriptorSetLayoutBinding> &layout_bindings,
+                                       std::vector<VkWriteDescriptorSet> &write_sets, const std::string &name)
+    : m_device(device), m_name(name), m_write_descriptor_sets(write_sets),
+      m_descriptor_set_layout_bindings(layout_bindings), m_swapchain_image_count(swapchain_image_count) {
+    assert(device);
+    assert(!layout_bindings.empty());
+    assert(swapchain_image_count > 0);
+    assert(!m_write_descriptor_sets.empty());
+    assert(layout_bindings.size() == m_write_descriptor_sets.size());
+
+    // Make sure descriptor writes are valid.
+    for (const auto &descriptor_write : m_write_descriptor_sets) {
+        // The target memory to which the descriptor write points must be valid!
+        assert(descriptor_write.pBufferInfo);
```

Comment:
```suggestion
        assert(descriptor_write.pBufferInfo != nullptr);
```

Reply:
This actually needs to be changed to `descriptor_write.pBufferInfo != nullptr || descriptor_write.pImageInfo != nullptr`, to allow for image descriptors.

Reply:
Done.